pkg/controller/bootstrap/bootstrapsigner.go

```diff
@@ -155,7 +155,7 @@ func NewSigner(cl clientset.Interface, secrets informers.SecretInformer, configM
 }
 
 // Run runs controller loops and returns when they are done
-func (e *Signer) Run(stopCh <-chan struct{}) {
+func (e *Signer) Run(ctx context.Context, stopCh <-chan struct{}) {
```

Comment:
taking both a context and stopCh is pretty confusing... shouldn't that collapse to just ctx and wait on `<-ctx.Done()`? same comment applies in lots of places

Reply:
Agreed, we should only use `context.Context` since that has information about stopCh coming from `ctx.Done()` invocation.